integration/data/conftest.py

```diff
@@ -49,6 +50,38 @@ def grpc_controller_no_frontend(request, grpc_controller_client):
                                   frontend="")
 
 
+@pytest.fixture
+def first_available_device(request):
+    # we find the next available scsi name based on the current
+    # nodes in the /dev filesystem which isn't fool proof
+    device_prefix = "/dev/sd"
+    first_available_device = "does_not_exist"
+    for i in list(string.ascii_lowercase):
+        dev = device_prefix + i
+        if not os.path.exists(dev):
+            first_available_device = dev
+            break
+
```

Comment:
Add `assert first_available_device != ""`

Reply:
See below

Reply:
Though this `assert` should fail the test immediately.

Reply:
fair point, will change it.